src/simplewallet/simplewallet.cpp

```diff
@@ -95,12 +95,7 @@ typedef cryptonote::simple_wallet sw;
   m_auto_refresh_enabled.store(false, std::memory_order_relaxed); \
   /* stop any background refresh, and take over */ \
   m_wallet->stop(); \
-  m_idle_mutex.lock(); \
-  while (m_auto_refresh_refreshing) \
-    m_idle_cond.notify_one(); \
-  m_idle_mutex.unlock(); \
```

Comment:
Damn. Right. Updated.

I knew that programming with threads is tricky, but not yet from firsthand experience, and this here really is quite humbling ...

Reply:
Now, since this is not testing whether we're in refresh anymore, it's probably better to call notify_all, to guard against more than one other thread locking m_idle_mutex. That guarantees all threads will have a go.